### PR TITLE
release/1.1

### PR DESCRIPTION
**Pull request by AlanGriffiths**
Release 1.1.2

### CODE REVIEW

**Comment by AlanGriffiths** [comment]
@Conan-Kudo, @z3ntu if there are any other fixes that could be included to help you please call them out.

**Comment by AlanGriffiths** [comment]
@RAOF seems that #733 doesn't work as well as MIR_MESA_KMS_DISABLE_MODESET_PROBE on pmOS. Any clues in the following?
```
qemu-amd64:~# 
qemu-amd64:~# export XDG_RUNTIME_DIR=/tmp/$(id -u)-runtime-dir
qemu-amd64:~# if ! test -d "${XDG_RUNTIME_DIR}"; then
>     mkdir "${XDG_RUNTIME_DIR}"
>     chmod 0700 "${XDG_RUNTIME_DIR}"
> fi
qemu-amd64:~# 
qemu-amd64:~# export LD_LIBRARY_PATH=/usr/lib/qt5/plugins/platforms
qemu-amd64:~# export MIR_SERVER_CURSOR=null
qemu-amd64:~# export QT_QPA_PLATFORM=mirserver
qemu-amd64:~# export G_MESSAGES_DEBUG=all
qemu-amd64:~# export QT_WAYLAND_DISABLE_WINDOWDECORATION=1
qemu-amd64:~# 
qemu-amd64:~# ck-launch-session unity8
[2019-02-06:09:22:53.628] qtmir.screens: ScreensModel::ScreensModel
[2019-02-06 09:22:53.674162] <DEBUG> mircommon: Cannot check "add_graphics_platform_options" symbol version is "MIR_GRAPHICS_PLATFORM_0.32": dlvsym() is unavailable
[2019-02-06 09:22:53.674580] <DEBUG> mircommon: Cannot check "add_graphics_platform_options" symbol version is "MIR_GRAPHICS_PLATFORM_0.32": dlvsym() is unavailable
[2019-02-06 09:22:53.674852] <DEBUG> mircommon: Cannot check "add_graphics_platform_options" symbol version is "MIR_GRAPHICS_PLATFORM_0.32": dlvsym() is unavailable
[2019-02-06 09:22:53.725780] <DEBUG> mircommon: Cannot check "add_graphics_platform_options" symbol version is "MIR_GRAPHICS_PLATFORM_0.32": dlvsym() is unavailable
[2019-02-06 09:22:53.726231] <DEBUG> mircommon: Cannot check "add_graphics_platform_options" symbol version is "MIR_GRAPHICS_PLATFORM_0.32": dlvsym() is unavailable
[2019-02-06 09:22:53.726936] mirserver: Starting
[2019-02-06 09:22:53.731712] <DEBUG> mirserver: Not using logind for session management: Failed to find active session
[2019-02-06 09:22:53.733512] <DEBUG> mirserver: Using Linux VT subsystem for session management
[2019-02-06 09:22:53.738271] mircommon: Loading modules from: /usr/lib/mir/server-platform
[2019-02-06 09:22:53.738610] mircommon: Loading module: /usr/lib/mir/server-platform/server-mesa-x11.so.16
[2019-02-06 09:22:53.738925] mircommon: Loading module: /usr/lib/mir/server-platform/graphics-mesa-kms.so.16
[2019-02-06 09:22:53.739204] mircommon: Loading module: /usr/lib/mir/server-platform/input-evdev.so.7
[2019-02-06 09:22:53.739477] mircommon: Loading module: /usr/lib/mir/server-platform/input-stub.so
[2019-02-06 09:22:53.739762] mircommon: Loading module: /usr/lib/mir/server-platform/graphics-dummy.so
[2019-02-06 09:22:53.739922] <DEBUG> mircommon: Cannot check "probe_graphics_platform" symbol version is "MIR_GRAPHICS_PLATFORM_0.32": dlvsym() is unavailable
[2019-02-06 09:22:53.740055] <DEBUG> mircommon: Cannot check "describe_graphics_module" symbol version is "MIR_GRAPHICS_PLATFORM_0.32": dlvsym() is unavailable
[2019-02-06 09:22:53.740204] mirplatform: Found graphics driver: mir:mesa-x11 (version 1.1.2) Support priority: 0
[2019-02-06 09:22:53.740449] <DEBUG> mircommon: Cannot check "probe_graphics_platform" symbol version is "MIR_GRAPHICS_PLATFORM_0.32": dlvsym() is unavailable
[2019-02-06 09:22:53.740758] mesa-kms: EGL platform does not support EGL_KHR_platform_gbm extension
[2019-02-06 09:22:53.741137] mesa-kms: Device /dev/dri/card0 does not support KMS
[2019-02-06 09:22:53.742333] <DEBUG> mircommon: Cannot check "describe_graphics_module" symbol version is "MIR_GRAPHICS_PLATFORM_0.32": dlvsym() is unavailable
[2019-02-06 09:22:53.742812] mirplatform: Found graphics driver: mir:mesa-kms (version 1.1.2) Support priority: 0
[2019-02-06 09:22:53.742829] <DEBUG> mircommon: Cannot check "probe_graphics_platform" symbol version is "MIR_GRAPHICS_PLATFORM_0.32": dlvsym() is unavailable
[2019-02-06 09:22:53.742880] <DEBUG> mircommon: Cannot check "probe_graphics_platform" symbol version is "MIR_GRAPHICS_PLATFORM_0.27": dlvsym() is unavailable
[2019-02-06 09:22:53.742945] <DEBUG> mircommon: Cannot check "probe_graphics_platform" symbol version is "MIR_GRAPHICS_PLATFORM_0.32": dlvsym() is unavailable
[2019-02-06 09:22:53.742978] <DEBUG> mircommon: Cannot check "probe_graphics_platform" symbol version is "MIR_GRAPHICS_PLATFORM_0.27": dlvsym() is unavailable
[2019-02-06 09:22:53.743006] <DEBUG> mircommon: Cannot check "probe_graphics_platform" symbol version is "MIR_GRAPHICS_PLATFORM_0.32": dlvsym() is unavailable
[2019-02-06 09:22:53.743020] <DEBUG> mircommon: Cannot check "describe_graphics_module" symbol version is "MIR_GRAPHICS_PLATFORM_0.32": dlvsym() is unavailable
[2019-02-06 09:22:53.743038] mirplatform: Found graphics driver: mir:stub-graphics (version 1.1.2) Support priority: 1
[2019-02-06 09:22:53.743053] <DEBUG> mircommon: Cannot check "create_host_platform" symbol version is "MIR_GRAPHICS_PLATFORM_0.32": dlvsym() is unavailable
[2019-02-06 09:22:53.743065] <DEBUG> mircommon: Cannot check "describe_graphics_module" symbol version is "MIR_GRAPHICS_PLATFORM_0.32": dlvsym() is unavailable
[2019-02-06 09:22:53.743077] mirserver: Selected driver: mir:stub-graphics (version 1.1.2)
[2019-02-06:09:22:53.744] qtmir.mir: PromptSessionListener::PromptSessionListener - this= PromptSessionListener(0x5590e5c50630)
[2019-02-06 09:22:53.744470] mirserver: Cursor disabled
[2019-02-06 09:22:53.757049] <DEBUG> mircommon: Cannot check "describe_input_module" symbol version is "MIR_INPUT_PLATFORM_0.27": dlvsym() is unavailable
[2019-02-06 09:22:53.757437] mircommon: Loading modules from: /usr/lib/mir/server-platform
[2019-02-06 09:22:53.757735] mircommon: Loading module: /usr/lib/mir/server-platform/server-mesa-x11.so.16
[2019-02-06 09:22:53.757989] <DEBUG> mircommon: Cannot check "probe_input_platform" symbol version is "MIR_INPUT_PLATFORM_0.27": dlvsym() is unavailable
[2019-02-06 09:22:53.758152] mircommon: Loading module: /usr/lib/mir/server-platform/graphics-mesa-kms.so.16
[2019-02-06 09:22:53.758321] <DEBUG> mircommon: Cannot check "probe_input_platform" symbol version is "MIR_INPUT_PLATFORM_0.27": dlvsym() is unavailable
[2019-02-06 09:22:53.758475] mircommon: Loading module: /usr/lib/mir/server-platform/input-evdev.so.7
[2019-02-06 09:22:53.758735] <DEBUG> mircommon: Cannot check "probe_input_platform" symbol version is "MIR_INPUT_PLATFORM_0.27": dlvsym() is unavailable
[2019-02-06 09:22:53.812756] <DEBUG> mircommon: Cannot check "describe_input_module" symbol version is "MIR_INPUT_PLATFORM_0.27": dlvsym() is unavailable
[2019-02-06 09:22:53.813715] <DEBUG> mircommon: Cannot check "create_input_platform" symbol version is "MIR_INPUT_PLATFORM_0.27": dlvsym() is unavailable
[2019-02-06 09:22:53.814458] mirserver: Selected input driver: mir:evdev-input (version: 1.1.2)
[2019-02-06:09:22:53.816] qtmir.mir: MirServer created
[2019-02-06:09:22:53.817] qtmir.mir: Command line arguments passed to Qt: ("unity8")
[2019-02-06 09:22:53.818152] mirserver: Mir version 1.1.2
[2019-02-06:09:22:53.818] qtmir.screens: QtCompositor::start
[2019-02-06 09:22:53.822382] mirserver: Initial display configuration:
[2019-02-06 09:22:53.822422] mirserver: * Output 1: VGA connected, used
[2019-02-06 09:22:53.822450] mirserver: . |_ Physical size 0.0" 0x0mm
[2019-02-06 09:22:53.822468] mirserver: . |_ Power is on
[2019-02-06 09:22:53.822484] mirserver: . |_ Current mode 1600x1600 60.00Hz
[2019-02-06 09:22:53.822500] mirserver: . |_ Preferred mode 1600x1600 60.00Hz
[2019-02-06 09:22:53.822513] mirserver: . |_ Orientation normal
[2019-02-06 09:22:53.822524] mirserver: . |_ Logical size 1600x1600
[2019-02-06 09:22:53.822537] mirserver: . |_ Logical position +0+0
[2019-02-06 09:22:53.822550] mirserver: . |_ Scaling factor: 1.00
[2019-02-06:09:22:53.822] qtmir.screens: ScreensModel::update
[2019-02-06:09:22:53.824] qtmir.sensor: Screen - nativeOrientation is: Qt::LandscapeOrientation
[2019-02-06:09:22:53.825] qtmir.sensor: Screen - initial currentOrientation is: Qt::LandscapeOrientation
[2019-02-06:09:22:53.843] qtmir.screens: Added Screen with id 1 and geometry QRect(0,0 1600x1600)
[2019-02-06:09:22:53.843] qtmir.screens: Screen::setMirDisplayBuffer Screen(0x5590e5c65100)
[2019-02-06:09:22:54.083] DisplayBuffer does not support GL rendering
qemu-amd64:~# 
```

**Comment by RAOF** [comment]
@AlanGriffiths Urgh, seriously?
The code-flow to get there is that `drmCheckModesettingSupported()` succeeds, returning `ENOSYS` (ie: we've successfully detected that the drm node does *not* support KMS).

Looking at the code, in the unlikely event that libdrm is not built with  the `__linux__` symbol defined¹ then `drmCheckModesettingSupported()` will always return `ENOSYS`. But (a) gcc (and clang) should always set `__linux__`, and (b) other parts of libdrm will be broken if `__linux__` is not defined…

Does that qemu system *actually* have KMS? If you remove the check, does it successfully start? How have you built that image, and can I have a copy?

¹: Or `__BSD__` or `__DRAGONFLYBSD__` or a host of other platform macros

**Comment by Conan-Kudo** [comment]
@RAOF I'm not sure things like that build properly under musl... They don't define much in the way of platform definitions... :/

**Comment by AlanGriffiths** [comment]
> Does that qemu system _actually_ have KMS? If you remove the check, does it successfully start? How have you built that image, and can I have a copy?

Sorry, all I have is a private message on Telegram from @z3ntu saying that the MIR_MESA_KMS_DISABLE_MODESET_PROBE workaround we removed stopped working.

**Comment by AlanGriffiths** [comment]
@z3ntu I've pushed a release/1.1-experimental branch intended to confirm our understanding of what you are seeing. It uses the same MIR_MESA_KMS_DISABLE_MODESET_PROBE environment variable to override the checks that are failing.

Could you try with and without that set and post the logs?

However, as [discussed above](https://github.com/MirServer/mir/pull/749#issuecomment-469080589), it does look as though the error is in the libdrm implementation you are using. If we have understood correctly what is happening, that would be the logical component to fix.

**Comment by z3ntu** [comment]
Here's the output with it set:
```
[2019-02-06:09:24:40.222] qtmir.screens: ScreensModel::ScreensModel
[2019-02-06 09:24:40.259752] <DEBUG> mircommon: Cannot check "add_graphics_platform_options" symbol version is "MIR_GRAPHICS_PLATFORM_0.32": dlvsym() is unavailable
[2019-02-06 09:24:40.260169] <DEBUG> mircommon: Cannot check "add_graphics_platform_options" symbol version is "MIR_GRAPHICS_PLATFORM_0.32": dlvsym() is unavailable
[2019-02-06 09:24:40.260506] <DEBUG> mircommon: Cannot check "add_graphics_platform_options" symbol version is "MIR_GRAPHICS_PLATFORM_0.32": dlvsym() is unavailable
[2019-02-06 09:24:40.301490] <DEBUG> mircommon: Cannot check "add_graphics_platform_options" symbol version is "MIR_GRAPHICS_PLATFORM_0.32": dlvsym() is unavailable
[2019-02-06 09:24:40.301785] <DEBUG> mircommon: Cannot check "add_graphics_platform_options" symbol version is "MIR_GRAPHICS_PLATFORM_0.32": dlvsym() is unavailable
[2019-02-06 09:24:40.302654] mirserver: Starting
[2019-02-06 09:24:40.305841] <DEBUG> mirserver: Not using logind for session management: Failed to find active session
[2019-02-06 09:24:40.306935] <DEBUG> mirserver: Using Linux VT subsystem for session management
[2019-02-06 09:24:40.307561] mircommon: Loading modules from: /usr/lib/mir/server-platform
[2019-02-06 09:24:40.307833] mircommon: Loading module: /usr/lib/mir/server-platform/graphics-mesa-kms.so.16
[2019-02-06 09:24:40.308008] mircommon: Loading module: /usr/lib/mir/server-platform/server-mesa-x11.so.16
[2019-02-06 09:24:40.308150] mircommon: Loading module: /usr/lib/mir/server-platform/input-evdev.so.7
[2019-02-06 09:24:40.308432] mircommon: Loading module: /usr/lib/mir/server-platform/graphics-dummy.so
[2019-02-06 09:24:40.308677] mircommon: Loading module: /usr/lib/mir/server-platform/input-stub.so
[2019-02-06 09:24:40.308854] <DEBUG> mircommon: Cannot check "probe_graphics_platform" symbol version is "MIR_GRAPHICS_PLATFORM_0.32": dlvsym() is unavailable
[2019-02-06 09:24:40.309251] mesa-kms: EGL platform does not support EGL_KHR_platform_gbm extension
[2019-02-06 09:24:40.309665] <WARNING> mesa-kms: Device /dev/dri/card0 does not support KMS: CHECK OVERRIDDEN, continuing with lower confidence
[2019-02-06 09:24:40.311791] <WARNING> mesa-kms: Unexpected error from drmCheckModesettingSupported()
[2019-02-06 09:24:40.312151] <WARNING> mesa-kms: Please file a bug at https://github.com/MirServer/mir/issues containing this message
[2019-02-06 09:24:40.312468] <WARNING> mesa-kms: drmCheckModesettingSupported() failed: Function not implemented (38)
[2019-02-06 09:24:40.312716] <WARNING> mesa-kms: Continuing probe with lower confidence
[2019-02-06 09:24:40.652251] <DEBUG> mircommon: Cannot check "describe_graphics_module" symbol version is "MIR_GRAPHICS_PLATFORM_0.32": dlvsym() is unavailable
[2019-02-06 09:24:40.652858] mirplatform: Found graphics driver: mir:mesa-kms (version 1.1.2) Support priority: 128
[2019-02-06 09:24:40.653437] <DEBUG> mircommon: Cannot check "probe_graphics_platform" symbol version is "MIR_GRAPHICS_PLATFORM_0.32": dlvsym() is unavailable
[2019-02-06 09:24:40.653890] <DEBUG> mircommon: Cannot check "describe_graphics_module" symbol version is "MIR_GRAPHICS_PLATFORM_0.32": dlvsym() is unavailable
[2019-02-06 09:24:40.654308] mirplatform: Found graphics driver: mir:mesa-x11 (version 1.1.2) Support priority: 0
[2019-02-06 09:24:40.654775] <DEBUG> mircommon: Cannot check "probe_graphics_platform" symbol version is "MIR_GRAPHICS_PLATFORM_0.32": dlvsym() is unavailable
[2019-02-06 09:24:40.655278] <DEBUG> mircommon: Cannot check "probe_graphics_platform" symbol version is "MIR_GRAPHICS_PLATFORM_0.27": dlvsym() is unavailable
[2019-02-06 09:24:40.655707] <DEBUG> mircommon: Cannot check "probe_graphics_platform" symbol version is "MIR_GRAPHICS_PLATFORM_0.32": dlvsym() is unavailable
[2019-02-06 09:24:40.655991] <DEBUG> mircommon: Cannot check "describe_graphics_module" symbol version is "MIR_GRAPHICS_PLATFORM_0.32": dlvsym() is unavailable
[2019-02-06 09:24:40.656146] mirplatform: Found graphics driver: mir:stub-graphics (version 1.1.2) Support priority: 1
[2019-02-06 09:24:40.656160] <DEBUG> mircommon: Cannot check "probe_graphics_platform" symbol version is "MIR_GRAPHICS_PLATFORM_0.32": dlvsym() is unavailable
[2019-02-06 09:24:40.656192] <DEBUG> mircommon: Cannot check "probe_graphics_platform" symbol version is "MIR_GRAPHICS_PLATFORM_0.27": dlvsym() is unavailable
[2019-02-06 09:24:40.656365] <DEBUG> mircommon: Cannot check "create_host_platform" symbol version is "MIR_GRAPHICS_PLATFORM_0.32": dlvsym() is unavailable
[2019-02-06 09:24:40.656380] <DEBUG> mircommon: Cannot check "describe_graphics_module" symbol version is "MIR_GRAPHICS_PLATFORM_0.32": dlvsym() is unavailable
[2019-02-06 09:24:40.656394] mirserver: Selected driver: mir:mesa-kms (version 1.1.2)
[2019-02-06 09:24:40.656703] <WARNING> mesa-kms: NOT Ignoring non-KMS DRM device /dev/dri/card0: CHECK OVERRIDDEN
[2019-02-06:09:24:40.657] Exception while creating graphics platform
ERROR: /home/pmos/build/src/mir-27d6b4bde55135363dfd3d70ff913f65270082a1/src/platforms/mesa/server/display_helpers.cpp(165): Throw in function static std::vector<std::shared_ptr<mgmh::DRMHelper> > mir::graphics::mesa::helpers::DRMHelper::open_all_devices(const std::shared_ptr<mir::udev::Context> &, mir::ConsoleServices &)
Dynamic exception type: boost::exception_detail::clone_impl<boost::exception_detail::error_info_injector<std::system_error> >
std::exception::what: Error opening DRM device: No error information
```
And without:
```
[2019-02-06:09:28:36.969] qtmir.screens: ScreensModel::ScreensModel
[2019-02-06 09:28:36.972961] <DEBUG> mircommon: Cannot check "add_graphics_platform_options" symbol version is "MIR_GRAPHICS_PLATFORM_0.32": dlvsym() is unavailable
[2019-02-06 09:28:36.973281] <DEBUG> mircommon: Cannot check "add_graphics_platform_options" symbol version is "MIR_GRAPHICS_PLATFORM_0.32": dlvsym() is unavailable
[2019-02-06 09:28:36.973467] <DEBUG> mircommon: Cannot check "add_graphics_platform_options" symbol version is "MIR_GRAPHICS_PLATFORM_0.32": dlvsym() is unavailable
[2019-02-06 09:28:37.007611] <DEBUG> mircommon: Cannot check "add_graphics_platform_options" symbol version is "MIR_GRAPHICS_PLATFORM_0.32": dlvsym() is unavailable
[2019-02-06 09:28:37.008790] <DEBUG> mircommon: Cannot check "add_graphics_platform_options" symbol version is "MIR_GRAPHICS_PLATFORM_0.32": dlvsym() is unavailable
[2019-02-06 09:28:37.009379] mirserver: Starting
[2019-02-06 09:28:37.011888] <DEBUG> mirserver: Not using logind for session management: Failed to find active session
[2019-02-06 09:28:37.012431] <DEBUG> mirserver: Using Linux VT subsystem for session management
[2019-02-06 09:28:37.012760] mircommon: Loading modules from: /usr/lib/mir/server-platform
[2019-02-06 09:28:37.013088] mircommon: Loading module: /usr/lib/mir/server-platform/graphics-mesa-kms.so.16
[2019-02-06 09:28:37.013354] mircommon: Loading module: /usr/lib/mir/server-platform/server-mesa-x11.so.16
[2019-02-06 09:28:37.013530] mircommon: Loading module: /usr/lib/mir/server-platform/input-evdev.so.7
[2019-02-06 09:28:37.013780] mircommon: Loading module: /usr/lib/mir/server-platform/graphics-dummy.so
[2019-02-06 09:28:37.013926] mircommon: Loading module: /usr/lib/mir/server-platform/input-stub.so
[2019-02-06 09:28:37.014204] <DEBUG> mircommon: Cannot check "probe_graphics_platform" symbol version is "MIR_GRAPHICS_PLATFORM_0.32": dlvsym() is unavailable
[2019-02-06 09:28:37.014605] mesa-kms: EGL platform does not support EGL_KHR_platform_gbm extension
[2019-02-06 09:28:37.015100] mesa-kms: Device /dev/dri/card0 does not support KMS
[2019-02-06 09:28:37.016066] <DEBUG> mircommon: Cannot check "describe_graphics_module" symbol version is "MIR_GRAPHICS_PLATFORM_0.32": dlvsym() is unavailable
[2019-02-06 09:28:37.016086] mirplatform: Found graphics driver: mir:mesa-kms (version 1.1.2) Support priority: 0
[2019-02-06 09:28:37.016096] <DEBUG> mircommon: Cannot check "probe_graphics_platform" symbol version is "MIR_GRAPHICS_PLATFORM_0.32": dlvsym() is unavailable
[2019-02-06 09:28:37.016117] <DEBUG> mircommon: Cannot check "describe_graphics_module" symbol version is "MIR_GRAPHICS_PLATFORM_0.32": dlvsym() is unavailable
[2019-02-06 09:28:37.016129] mirplatform: Found graphics driver: mir:mesa-x11 (version 1.1.2) Support priority: 0
[2019-02-06 09:28:37.016138] <DEBUG> mircommon: Cannot check "probe_graphics_platform" symbol version is "MIR_GRAPHICS_PLATFORM_0.32": dlvsym() is unavailable
[2019-02-06 09:28:37.016172] <DEBUG> mircommon: Cannot check "probe_graphics_platform" symbol version is "MIR_GRAPHICS_PLATFORM_0.27": dlvsym() is unavailable
[2019-02-06 09:28:37.016198] <DEBUG> mircommon: Cannot check "probe_graphics_platform" symbol version is "MIR_GRAPHICS_PLATFORM_0.32": dlvsym() is unavailable
[2019-02-06 09:28:37.016210] <DEBUG> mircommon: Cannot check "describe_graphics_module" symbol version is "MIR_GRAPHICS_PLATFORM_0.32": dlvsym() is unavailable
[2019-02-06 09:28:37.016249] mirplatform: Found graphics driver: mir:stub-graphics (version 1.1.2) Support priority: 1
[2019-02-06 09:28:37.016258] <DEBUG> mircommon: Cannot check "probe_graphics_platform" symbol version is "MIR_GRAPHICS_PLATFORM_0.32": dlvsym() is unavailable
[2019-02-06 09:28:37.016282] <DEBUG> mircommon: Cannot check "probe_graphics_platform" symbol version is "MIR_GRAPHICS_PLATFORM_0.27": dlvsym() is unavailable
[2019-02-06 09:28:37.016309] <DEBUG> mircommon: Cannot check "create_host_platform" symbol version is "MIR_GRAPHICS_PLATFORM_0.32": dlvsym() is unavailable
[2019-02-06 09:28:37.016320] <DEBUG> mircommon: Cannot check "describe_graphics_module" symbol version is "MIR_GRAPHICS_PLATFORM_0.32": dlvsym() is unavailable
[2019-02-06 09:28:37.016331] mirserver: Selected driver: mir:stub-graphics (version 1.1.2)
[2019-02-06:09:28:37.016] qtmir.mir: PromptSessionListener::PromptSessionListener - this= PromptSessionListener(0x562684bc3030)
[2019-02-06 09:28:37.016594] mirserver: Cursor disabled
[2019-02-06 09:28:37.026620] <DEBUG> mircommon: Cannot check "describe_input_module" symbol version is "MIR_INPUT_PLATFORM_0.27": dlvsym() is unavailable
[2019-02-06 09:28:37.027048] mircommon: Loading modules from: /usr/lib/mir/server-platform
[2019-02-06 09:28:37.027376] mircommon: Loading module: /usr/lib/mir/server-platform/graphics-mesa-kms.so.16
[2019-02-06 09:28:37.027665] <DEBUG> mircommon: Cannot check "probe_input_platform" symbol version is "MIR_INPUT_PLATFORM_0.27": dlvsym() is unavailable
[2019-02-06 09:28:37.027930] mircommon: Loading module: /usr/lib/mir/server-platform/server-mesa-x11.so.16
[2019-02-06 09:28:37.028120] <DEBUG> mircommon: Cannot check "probe_input_platform" symbol version is "MIR_INPUT_PLATFORM_0.27": dlvsym() is unavailable
[2019-02-06 09:28:37.028261] mircommon: Loading module: /usr/lib/mir/server-platform/input-evdev.so.7
[2019-02-06 09:28:37.028540] <DEBUG> mircommon: Cannot check "probe_input_platform" symbol version is "MIR_INPUT_PLATFORM_0.27": dlvsym() is unavailable
[2019-02-06 09:28:37.074848] <DEBUG> mircommon: Cannot check "describe_input_module" symbol version is "MIR_INPUT_PLATFORM_0.27": dlvsym() is unavailable
[2019-02-06 09:28:37.075476] <DEBUG> mircommon: Cannot check "create_input_platform" symbol version is "MIR_INPUT_PLATFORM_0.27": dlvsym() is unavailable
[2019-02-06 09:28:37.076022] mirserver: Selected input driver: mir:evdev-input (version: 1.1.2)
[2019-02-06:09:28:37.077] qtmir.mir: MirServer created
[2019-02-06:09:28:37.078] qtmir.mir: Command line arguments passed to Qt: ("unity8")
[2019-02-06 09:28:37.078632] mirserver: Mir version 1.1.2
[2019-02-06:09:28:37.079] qtmir.screens: QtCompositor::start
[2019-02-06 09:28:37.081410] mirserver: Initial display configuration:
[2019-02-06 09:28:37.081433] mirserver: * Output 1: VGA connected, used
[2019-02-06 09:28:37.081453] mirserver: . |_ Physical size 0.0" 0x0mm
[2019-02-06 09:28:37.081465] mirserver: . |_ Power is on
[2019-02-06 09:28:37.081477] mirserver: . |_ Current mode 1600x1600 60.00Hz
[2019-02-06 09:28:37.081488] mirserver: . |_ Preferred mode 1600x1600 60.00Hz
[2019-02-06 09:28:37.081498] mirserver: . |_ Orientation normal
[2019-02-06 09:28:37.081508] mirserver: . |_ Logical size 1600x1600
[2019-02-06 09:28:37.081519] mirserver: . |_ Logical position +0+0
[2019-02-06 09:28:37.081529] mirserver: . |_ Scaling factor: 1.00
[2019-02-06:09:28:37.081] qtmir.screens: ScreensModel::update
libinput error: [2019-02-06:09:28:37.083] qtmir.sensor: Screen - nativeOrientation is: Qt::LandscapeOrientation
[2019-02-06:09:28:37.084] qtmir.sensor: Screen - initial currentOrientation is: Qt::LandscapeOrientation
/usr/share/libinput: failed to find data files
libinput error: Failed to load the device quirks from /usr/share/libinput and /etc/libinput/local-overrides.quirks. This will negatively affect device behavior. See https://wayland.freedesktop.org/libinput/doc/1.12.6/device-quirks.html for details.
[2019-02-06:09:28:37.093] qtmir.screens: Added Screen with id 1 and geometry QRect(0,0 1600x1600)
[2019-02-06:09:28:37.093] qtmir.screens: Screen::setMirDisplayBuffer Screen(0x562684ba3f00)
[2019-02-06:09:28:37.325] DisplayBuffer does not support GL rendering
```
The full QEMU command being used is the following, if that helps
```
QEMU_MODULE_DIR=/home/luca/.local/var/pmbootstrap/chroot_native/usr/lib/qemu GBM_DRIVERS_PATH=/home/luca/.local/var/pmbootstrap/chroot_native/usr/lib/xorg/modules/dri LIBGL_DRIVERS_PATH=/home/luca/.local/var/pmbootstrap/chroot_native/usr/lib/xorg/modules/dri /home/luca/.local/var/pmbootstrap/chroot_native/lib/ld-musl-x86_64.so.1 --library-path=/home/luca/.local/var/pmbootstrap/chroot_native/lib:/home/luca/.local/var/pmbootstrap/chroot_native/usr/lib /home/luca/.local/var/pmbootstrap/chroot_native/usr/bin/qemu-system-x86_64 -L /home/luca/.local/var/pmbootstrap/chroot_native/usr/share/qemu/ -kernel /home/luca/.local/var/pmbootstrap/chroot_rootfs_qemu-amd64/boot/vmlinuz-postmarketos-stable -initrd /home/luca/.local/var/pmbootstrap/chroot_rootfs_qemu-amd64/boot/initramfs-postmarketos-stable -append 'console=tty1 console=ttyS0 PMOS_NO_OUTPUT_REDIRECT PMOS_FORCE_PARTITION_RESIZE' -m 1024 -netdev user,id=net0,hostfwd=tcp::2222-:22,hostfwd=tcp::2223-:23,net=172.16.42.0/24,dhcpstart=172.16.42.1 -show-cursor -serial stdio -drive file=/home/luca/.local/var/pmbootstrap/chroot_native/home/pmos/rootfs/qemu-amd64.img,format=raw -device e1000,netdev=net0 -smp 8 -enable-kvm -vga virtio -display sdl,gl=on
```

**Comment by AlanGriffiths** [comment]
@z3ntu I assume from the first trace that Unity8 still failed to start?

That's odd as, apart from making some checks it then ignores, it ought to be doing the same as Mir 1.1.1.

Could you double check that Mir 1.1.1 runs in this same environment? (With the MIR_MESA_KMS_DISABLE_MODESET_PROBE environment variable that is.)

**Comment by z3ntu** [comment]
Mir 1.1.1 with MIR_MESA_KMS_DISABLE_MODESET_PROBE=1 set: https://pastebin.com/raw/8Wa3Z0aV

**Comment by AlanGriffiths** [comment]
@z3ntu apologies. I got my experimental code wrong. Could you please retry with the latest commit?

**Comment by z3ntu** [comment]
Yep that works, log is here: https://pastebin.com/raw/qEGrazDV

**Comment by z3ntu** [comment]
So it's a bug in libdrm..?

**Comment by AlanGriffiths** [comment]
@z3ntu I've pushed a preview of this change to https://github.com/MirServer/mir/tree/release/1.1-preview

**Comment by z3ntu** [comment]
I've applied the patch manually in the build process (didn't see your comment until now) but the results should be the same - and it works fine.

**Comment by AlanGriffiths** [comment]
Tested on Ubuntu 18.04

**Comment by AlanGriffiths** [comment]
Tested on Ubuntu 19.04

**Comment by AlanGriffiths** [comment]
Tested on Ubuntu 18.10

**Comment by AlanGriffiths** [comment]
@wmww @RAOF

Could one of you confirm this works with Nvidia, and tag the release?

(Step 6 in https://discourse.ubuntu.com/t/mir-release-protocol/2519)

**Comment by AlanGriffiths** [comment]
@z3ntu I expect to tag the release later today. I know you've tested the equivalent patch, but would you care to recheck with the latest commit?

File: .travis.yml [hunk 1]
```diff
@@ -44,7 +44,7 @@ env:
   - SPREAD_PATH=/tmp/spread
   matrix:
   - SYSTEM=google:ubuntu-devel-64 VARIANT=amd64
-  - SYSTEM=google:fedora-rawhide-64 VARIANT=amd64
+#  - SYSTEM=google:fedora-rawhide-64 VARIANT=amd64 Getting GPG errors from dnf today
```
Comment (Conan-Kudo) [inline .travis.yml]: rawhide changed over to Fedora 31 keys last week, so your rawhide VM probably needs rebootstrapping.

**Comment by gerboland** [comment]
I tested this with Nvidia driver 4.15 on Bionic, works fine

**Comment by z3ntu** [comment]
Commit 7be556cc89d60ac68bfac92dd53dc7fffa3eeccf works fine on pmOS.

**Comment by AlanGriffiths** [comment]
@gerboland , @z3ntu thanks!

Tag applied: the release magic starts...

**Comment by RAOF** [comment]
Tested with nvidia 410 on Disco; good to go.

**Comment by AlanGriffiths** [comment]
Sanity checked 18.04 on X11, mesa/lvt & mesa/logind

**Comment by z3ntu** [comment]
Will 1.1.2 also get a .tar.xz attached on github with wlcs included?

**Comment by AlanGriffiths** [comment]
> Will 1.1.2 also get a .tar.xz attached on github with wlcs included?

Here you are: https://github.com/MirServer/mir/releases/tag/v1.1.2

**Comment by z3ntu** [comment]
Thanks!

**Pull request merged**